features/step-definitions/task.2.js

```diff
@@ -0,0 +1,143 @@
+
+const { Given, When, Then } = require('@cucumber/cucumber');
+const HomePage = require('../pageobjects/home.page');
+const HoverPage = require('../pageobjects/hover.page');
+const AlertPage=require('../pageobjects/alert.page');
+const keyPage = require('../pageobjects/key.page');
+let send_text=""
+const image_number=2;
+const Randomizing=require('../../Utils/random.text') ;
+
+//test case 3
+
+When(/^I click the hovers link$/,async() => {
+    await HomePage.ClickHoverLink()
+});
+
+Then(/^I am on the hovers page$/, async() => {
+	await expect(HoverPage.locateUserImage(image_number)).toBeExisting()
+   
+});
+
+When(/^I Hover over the profile image of user2$/, async() => {
+    
+    await HoverPage.locateUserImage(image_number).moveTo();
+   
+});
+
+Then(/^I see name: user 2 and view profile link are displayed under the image on hover$/, async() => {
+    await expect(HoverPage.locateUserProfile(image_number)).toBeDisplayed()
+	
+});
+
+
+
+
+//test case 4
+
+
+When(/^I click on the key press link$/, async() => {
+	
+    await HomePage.ClickKeyPageLink()
+});
+
+Then(/^I am on the Key Press page$/, async() => {
+	
+    await expect(keyPage.TextField).toBeExisting()
+});
+
+When(/^I press Backspace key$/, async() => {
+    
+    await browser.keys(['Backspace'])
+});
+
+Then(/^I can see the message You entered: BACK_SPACE$/,async () => {
+	
+    await expect(keyPage.ResultField).toHaveTextContaining('You entered: BACK_SPACE')
+});
+
+
+
+//test case 5
+
+
+
+When(/^I click on the JavaScript Alerts link$/, async() => {
+    await HomePage.ClickJSAlertLink()
+});
+
+Then(/^I am on the JavaScript Alerts page$/, async() => {
+	
+    await expect(AlertPage.JSAlert).toBeExisting()

```

Comment:
You should work with WDIO elements inside a page class, so you'll need a method e.g. isExists inside AlertPage class which returns true if element exists and false if it doesn't. And this method should be called in a step definition instead of working with an element directly